src/main/java/org/candlepin/config/IntegerConfigurationValidator.java

```diff
@@ -0,0 +1,217 @@
+        this.min = min;
+        this.max = max;
+        return this;
+    }
+
+    /**
+     * Generates validation criteria specifying that the configuration should be equal to or less than the
+     * configuration value corresponding to provided key. If the provided configuration key corresponds to a
+     * non-integer value, it results in a failure. Successive calls to this method append extra criteria to
+     * this validator that must be satisfied.
+     *
+     * @param key
+     *  a key used to retrieve a configuration value to be compared
+     *
+     * @return
+     *  a reference to this integer configuration validator
+     */
+    public IntegerConfigurationValidator lessThan(String key) {
+        if (key == null || key.isBlank()) {
+            throw new IllegalArgumentException("null or blank key");
+        }
+
+        this.lessThanKeys.add(key);
+        return this;
+    }
+
+    /**
+     * Generates validation criteria specifying that the configuration should be equal to or greater than the
+     * configuration value corresponding to provided key. If the provided configuration key corresponds to a
+     * non-integer value, it results in a failure. Successive calls to this method append extra criteria to
+     * this validator that must be satisfied.
+     *
+     * @param key
+     *  the configuration key used to retrieve a value to be compared against this configuration
+     *
+     * @return
+     *  a reference to this integer configuration validator
+     */
+    public IntegerConfigurationValidator greaterThan(String key) {
+        if (key == null || key.isBlank()) {
+            throw new IllegalArgumentException("null or blank key");
+        }
+
+        this.greaterThanKeys.add(key);
+        return this;
+    }
+
+    @Override
+    public Set<String> validate(Configuration config) {
+        String key = getKey();
+        Integer value = getInt(key, config);
+        if (value == null) {
+            return this.getFailures();
+        }
+
+        if (min != null && value < min) {
+            this.addFailure("Configuration \"" + key + "\" must be larger than " + min);
+        }
+
+        if (max != null && value > max) {
+            this.addFailure("Configuration \"" + key + "\" must be smaller than " + max);
+        }
+
+        validateLessThanConfigs(value, config);
+
+        validateGreaterThanConfigs(value, config);
+
+        return this.getFailures();
+    }
+
+    private void validateLessThanConfigs(Integer value, Configuration config) {
+        if (value == null) {
+            return;
+        }
+
+        if (!lessThanKeys.isEmpty()) {
+            for (String lessThanKey : lessThanKeys) {
+                Integer otherValue = getInt(lessThanKey, config);
+                if (otherValue != null && value > otherValue) {
+                    this.addFailure("Configuration \"" + getKey() +
+                        "\" must be smaller than \"" + lessThanKey + "\"");
```

Comment:
This (and the greaterThan method as well) can be slightly improved if it was `Configuration X must be smaller than the value of configuration Y`.

Reply:
Good catch, I think that makes the message more clear. Updated in both methods